benchmark_runner/main/main.py

```diff
@@ -169,9 +169,10 @@ def upgrade_ocp_bare_metal(step: str):
     if step == 'run_bare_metal_ocp_upgrade':
         if not bare_metal_operations.is_ocp_already_upgraded(oc):
             bare_metal_operations.run_ocp_upgrade(oc)
-            # The LSO/ODF upgrade must be run manually after the OCP upgrade for the channel version; it won't upgrade automatically.
-            bare_metal_operations.install_ocp_resources(resources=['lso'], upgrade_version=lso_version)
-            bare_metal_operations.install_ocp_resources(resources=['odf'], upgrade_version=odf_version)
+            # Manual LSO/ODF upgrade works up to version 4.17
+            if '.'.join(upgrade_ocp_version.split('.')[:2]) <= "4.17":
```

Comment:
This won't work correctly for e. g. 4.9.  You need to compare each component of the version.